src/pkcs11/core_pkcs11_test.c

```diff
@@ -0,0 +1,2647 @@
+    {
+        TEST_ASSERT_TRUE( 0 != ( CKF_SIGN & MechanismInfo.flags ) );
+
+        TEST_ASSERT_TRUE( MechanismInfo.ulMaxKeySize >= pkcs11RSA_2048_MODULUS_BITS &&
+                          MechanismInfo.ulMinKeySize <= pkcs11RSA_2048_MODULUS_BITS );
+
+        /* Check for pre-padded signature verification support. This is required
+         * for round-trip testing. */
+        xResult = pxGlobalFunctionList->C_GetMechanismInfo( xSlotId, CKM_RSA_X_509, &MechanismInfo );
+        TEST_ASSERT_TRUE( CKR_OK == xResult );
+
+        TEST_ASSERT_TRUE( 0 != ( CKF_VERIFY & MechanismInfo.flags ) );
+
+        TEST_ASSERT_TRUE( MechanismInfo.ulMaxKeySize >= pkcs11RSA_2048_MODULUS_BITS &&
+                          MechanismInfo.ulMinKeySize <= pkcs11RSA_2048_MODULUS_BITS );
+
+        /* Check consistency with static configuration. */
+        #if ( 0 == PKCS11_TEST_RSA_KEY_SUPPORT )
+            TEST_FAIL_MESSAGE( "Static and runtime configuration for key generation support are inconsistent." );
+        #endif
+
+        TEST_PRINTF( "%s", "The PKCS #11 module supports RSA signing.\r\n" );
+    }
+
+    /* Check for ECDSA support, if applicable. */
+    xResult = pxGlobalFunctionList->C_GetMechanismInfo( xSlotId, CKM_ECDSA, &MechanismInfo );
+    TEST_ASSERT_TRUE( CKR_OK == xResult || CKR_MECHANISM_INVALID == xResult );
+
+    if( CKR_OK == xResult )
+    {
+        TEST_ASSERT_TRUE( 0 != ( ( CKF_SIGN | CKF_VERIFY ) & MechanismInfo.flags ) );
+
+        TEST_ASSERT_TRUE( MechanismInfo.ulMaxKeySize >= pkcs11ECDSA_P256_KEY_BITS &&
+                          MechanismInfo.ulMinKeySize <= pkcs11ECDSA_P256_KEY_BITS );
+
+        /* Check consistency with static configuration. */
+        #if ( 0 == PKCS11_TEST_EC_KEY_SUPPORT )
+            TEST_FAIL_MESSAGE( "Static and runtime configuration for key generation support are inconsistent." );
+        #endif
+
+        TEST_PRINTF( "%s", "The PKCS #11 module supports ECDSA.\r\n" );
+    }
+
+    #if ( PKCS11_TEST_PREPROVISIONED_SUPPORT != 1 )
+        /* Check for elliptic-curve key generation support. */
+        xResult = pxGlobalFunctionList->C_GetMechanismInfo( xSlotId, CKM_EC_KEY_PAIR_GEN, &MechanismInfo );
+        TEST_ASSERT_TRUE( CKR_OK == xResult || CKR_MECHANISM_INVALID == xResult );
+
+        if( CKR_OK == xResult )
+        {
+            TEST_ASSERT_TRUE( 0 != ( CKF_GENERATE_KEY_PAIR & MechanismInfo.flags ) );
+
+            TEST_ASSERT_TRUE( MechanismInfo.ulMaxKeySize >= pkcs11ECDSA_P256_KEY_BITS &&
+                              MechanismInfo.ulMinKeySize <= pkcs11ECDSA_P256_KEY_BITS );
+
+            xSupportsKeyGen = CK_TRUE;
+            TEST_PRINTF( "%s", "The PKCS #11 module supports elliptic-curve key generation.\r\n" );
+        }
+
+        /* Check for consistency between static configuration and runtime key
+         * generation settings. */
+        if( CK_TRUE == xSupportsKeyGen )
+        {
+            #if ( 0 == PKCS11_TEST_GENERATE_KEYPAIR_SUPPORT )

```

Comment:
prvProvisionEcTestCredentials is used for other test cases need PKCS11 object already provisioned in the module. There are separate test cases for generated keypair and imported key. I would like to update the test so that all test cases will run on imported private key and key pair generation.